apps/minifront/src/state/staking.ts

```diff
@@ -162,26 +169,67 @@ export const createStakingSlice = (): SliceCreator<StakingSlice> => (set, get) =
     set(state => {
       state.staking.delegationsByAccount.set(addressIndex.account, []);
       state.staking.votingPowerByValidatorInfo = {};
+      state.staking.loading = true;
     });
 
-    for await (const delegation of getDelegationsForAccount(addressIndex)) {
-      const delegations = get().staking.delegationsByAccount.get(addressIndex.account) ?? [];
+    let delegationsToFlush: ValueView[] = [];
+    let allDelegationsLoaded = false;
 
-      const sortedDelegations = [...delegations, delegation].sort(byBalanceAndVotingPower);
+    /**
+     * Per the RPC call, we get delegations in a stream, one-by-one. If we push
+     * them to state as we receive them, React has to re-render super
+     * frequently. Rendering happens synchronously, which means that the `for`
+     * loop below has to wait until rendering is done before moving on to the
+     * next delegation. Thus, the staking page loads super slowly if we render
+     * delegations as soon as we receive them.
+     *
+     * To resolve this performance issue, we instead queue up a number of
+     * delegations (up to `DELEGATIONS_FLUSH_TO_STATE_INTERVAL`), and then flush
+     * them to state in batches.
+     */
+    const flushToState = (ignoreInterval = false) => {
+      if (allDelegationsLoaded) return;
 
-      set(state => {
-        state.staking.delegationsByAccount.set(addressIndex.account, sortedDelegations);
-      });
+      if (delegationsToFlush.length >= DELEGATIONS_FLUSH_TO_STATE_INTERVAL || ignoreInterval) {
+        const delegations = get().staking.delegationsByAccount.get(addressIndex.account) ?? [];
+
+        const sortedDelegations = [...delegations, ...delegationsToFlush].sort(
+          byBalanceAndVotingPower,
+        );
 
+        set(state => {
+          state.staking.delegationsByAccount.set(addressIndex.account, sortedDelegations);
+        });
+
+        delegationsToFlush = [];
+      }
+
+      requestAnimationFrame(() => flushToState());
```

Comment:
Oh interesting. I've only seen `requestAnimationFrame` in use when dealing with wanting to paint to the screen no faster than the refresh rate of the device. Regardless, this is still quite fast. Given there is `DELEGATIONS_FLUSH_TO_STATE_INTERVAL`, shouldn't that be the thing that controls the frequency of the re-render? It seems we have two levers that control throttling. What happens if you remove `requestAnimationFrame`?